libopae/src/reconf.c

```diff
@@ -134,6 +135,68 @@ static fpga_result validate_bitstream(fpga_handle handle,
 }
 
 
+// open child accelerator exclusively - it not, it's busy!
+static fpga_result open_accel(fpga_handle handle, fpga_handle *accel)
+{
+	fpga_result result                = FPGA_OK;
+	struct _fpga_handle *_handle      = (struct _fpga_handle *)handle;
+	fpga_token token;
+	fpga_properties props;
+	uint32_t matches = 0;
+
+	if (_handle == NULL) {
+		FPGA_ERR("Invalid handle");
+		return FPGA_INVALID_PARAM;
+	}
+
+	if (_handle->token == NULL) {
+		FPGA_ERR("Invalid token within handle");
+		return FPGA_INVALID_PARAM;
+	}
+
+	result = fpgaGetProperties(NULL, &props);
+	if (result != FPGA_OK)
+		return result;
+
+	result = fpgaPropertiesSetParent(props, _handle->token);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Error setting parent in properties.");
+		goto free_props;
+	}
+
+	// TODO: Use slot number as part of filter
+	//       We only want to query for accelerators for the
+	//       slot being reconfigured
+	result = fpgaEnumerate(&props, 1, &token, 1, &matches);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Error enumerating for accelerator to reconfigure");
+		goto free_props;
+	}
+
+	if (matches == 0) {
+		FPGA_ERR("No accelerator found to reconfigure");
+		result = FPGA_BUSY;
+		goto destroy_token;
+	}
+
+	result = fpgaOpen(token, accel, 0);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Could not open accelerator for given slot");
+		goto destroy_token;
+	}
+
+destroy_token:
+	result = fpgaDestroyToken(&token);
```

Comment:
This overwrites the result of `fpgaOpen()` in line 182, which may mask a failure and return FPGA_OK to the caller.